src/embind/emval.js

```diff
@@ -72,12 +72,13 @@ var LibraryEmVal = {
 
     toHandle: (value) => {
       switch (value) {
-        case undefined: return 1;
-        case null: return 2;
-        case true: return 3;
-        case false: return 4;
+        case undefined: return {{{ to64(1) }}};
+        case null: return {{{ to64(2) }}};
+        case true: return {{{ to64(3) }}};
+        case false: return {{{ to64(4) }}};
         default:{
-          return emval_handles.allocate({refcount: 1, value: value});
```

Comment:
I wonder if EM_VAL should not be changed from `struct _EM_VAL*` to  `int` (or even `double` to match the JS number type).

There is no need to pointer value here as far as I can tell.

Reply:
Heh, that's the first change I thought of, but I didn't want to lose opaque typechecking. So then I wrapped it into a custom struct, which is guaranteed by C ABI to be representable as int itself, but then realised this will make handles way too different from all other handles in Emscripten which are represented by opaque pointers.

So I figured it's better to do minimal change here first, and figure out general solutions for all handles later.